File: codingbat/java/centeredAverage.java
public int centeredAverage(int[] nums) {
  int minNum = nums[0];
  int maxNum = nums[0];
  int sumNum = 0;
  for (int num: nums) {
    if (minNum > num) minNum = num;
    if (maxNum < num) maxNum = num;
    sumNum += num;
  }
  sumNum -= (minNum + maxNum);
  return sumNum / (nums.length - 2);
}

File: codingbat/python/centered_average.py
def centered_average(nums):
  sum = 0
  minNum = nums[0]
  maxNum = nums[0]
  for num in nums:
    if minNum > num: minNum = num
    if maxNum < num: maxNum = num
    sum += num
  return (sum - minNum - maxNum) / (len(nums) - 2)

